src/lib/support/Pool.h

```diff
@@ -166,6 +167,16 @@ struct HeapObjectList : HeapObjectListNode
 
 } // namespace internal
 
+/**
+ * Action taken if objects remain allocated when a pool is destroyed.
+ */
+enum class OnObjectPoolDestruction
+{
+    AutoRelease,                   ///< Release any objects still allocated.
```

Comment:
I'm of two minds about including this — an ObjectPool user can just call `ReleaseAll()` in their destructor, and without it, we could remove this parameter and unconditionally fail, if/when existing uses are addressed.

Reply:
I think long-term explicit lifetime is easier to understand and enforce. Otherwise everyone will `AutoRelease` because less  code  to  write.

Reply:
Then again probably does not need to be done in  this PR (or at all). I  stand by  my checkmark of approval for this.

Reply:
I will definitely revisit this topic to remove the `IgnoreUnsafeDoNotUseInNewCode` case, and if there are no compelling use cases for `AutoRelease` by then I'll remove that as well.